pkg/k8shandler/cluster.go

```diff
@@ -47,183 +46,150 @@ func (elasticsearchRequest *ElasticsearchRequest) CreateOrUpdateElasticsearchClu
-				if elasticsearchRequest.AnyNodeReady() {
-					if ok, err := esClient.SetShardAllocation(api.ShardAllocationAll); !ok {
-						logrus.Warnf("Unable to enable shard allocation for cluster %q in namespace %q: %v", elasticsearchRequest.cluster.Name, elasticsearchRequest.cluster.Namespace, err)
-					}
-				}
+			if err := node.create(); err != nil {
+				return err
+			}
 
-				// we only want to update our replicas if we aren't in the middle up an update
-				if elasticsearchRequest.ClusterReady() {
-					if err := esClient.UpdateReplicaCount(int32(calculateReplicaCount(elasticsearchRequest.cluster))); err != nil {
-						logrus.Error(err)
-					}
-					if aliasNeededMap == nil {
-						aliasNeededMap = make(map[string]bool)
-					}
-					if val, ok := aliasNeededMap[nodeMapKey(elasticsearchRequest.cluster.Name, elasticsearchRequest.cluster.Namespace)]; !ok || val {
-						// add alias to old indices if they exist and don't have one
-						// this should be removed after one release...
-						successful := esClient.AddAliasForOldIndices()
-
-						if successful {
-							aliasNeededMap[nodeMapKey(elasticsearchRequest.cluster.Name, elasticsearchRequest.cluster.Namespace)] = false
-						}
-					}
-				}
+			addNodeState(node, nodeStatus)
+			if err := er.setNodeStatus(node, nodeStatus, clusterStatus); err != nil {
+				logrus.Warnf("unable to set status for node %q: %s", node.name(), err.Error())
 			}
 		}
-	}
-
-	// Scrape cluster health from elasticsearch every time
-	return elasticsearchRequest.UpdateClusterStatus()
-}
 
-func (elasticsearchRequest *ElasticsearchRequest) updateMinMasters() {
-	// do as best effort -- whenever we create a node update min masters (if required)
+		// ensure that MinMasters is (n / 2 + 1)
+		er.updateMinMasters()
 
-	cluster := elasticsearchRequest.cluster
-	esClient := elasticsearchRequest.esClient
+		// ensure we always have shard allocation to All if we aren't doing an update...
+		er.ensureAllShardAllocation()
 
-	currentMasterCount, err := esClient.GetMinMasterNodes()
-	if err != nil {
-		logrus.Debugf("Unable to get current min master count for cluster %q in namespace %q", cluster.Name, cluster.Namespace)
-	}
+		// we only want to update our replicas if we aren't in the middle up an update
+		er.updateReplicas()
 
-	desiredMasterCount := getMasterCount(cluster)/2 + 1
-	currentNodeCount, err := esClient.GetClusterNodeCount()
-	if err != nil {
-		logrus.Warnf("unable to get cluster node count for cluster %q in namespace %q: %s", cluster.Name, cluster.Namespace, err.Error())
-	}
+		// add alias to old indices if they exist and don't have one
+		// this should be removed after one release...
+		if er.ClusterReady() {
+			if aliasNeededMap == nil {
+				aliasNeededMap = make(map[string]bool)
+			}
+			if val, ok := aliasNeededMap[nodeMapKey(er.cluster.Name, er.cluster.Namespace)]; !ok || val {
+				successful := esClient.AddAliasForOldIndices()
```

Comment:
Is `AddAliasForOldIndices` shallowing any important error for the log?

Reply:
the function prints out its warnings, IMO we don't need to have any additional log output when it isn't successful

Reply:
Ok, if the log is produced on the callee side.